modules/mobile/app/controllers/mobile/v0/messages_controller.rb

```diff
@@ -115,6 +116,30 @@ def signature
 
       private
 
+      def log_bad_sort(resource)
+        last_sent_date = Time.now.utc
+        bad_sort_flag = false
+        nil_sent_dates_count = 0
+        sent_dates = resource.attributes.map do |message|
+          unless message&.sent_date
+            nil_sent_dates_count += 1
+            next
+          end
+
+          bad_sort_flag ||= message.sent_date > last_sent_date
+
+          last_sent_date = message.sent_date
```

Comment:
Are the messages coming in already sorted by `sent_date`? It seems the messages can be sorted by other fields which would make this code change behavior if I'm not mistaken? Since `last_sent_date` will be set to the previous message from the loop

Reply:
Yeah they theoretically should be coming in already sorted by sent_date. I've confirmed with FE that they never use a sort parameter and the model has some logic where if there's no sort parameter it'll default to sorting by sent_date. I couldn't find any issue with the code so hoping to catch some unique edge here. This logic wouldn't make sense if FE ever did give a sort parameter for a different field but don't think there's any plan for that. We'll remove this log after investigation too. Since there is so much logic for a log, I also added that rescue log just in case there was an error, i didn't want it to cause issue with the endpoint.

Reply:
Cool, works for me. I was only worried the logs would miss logging what you want if the orders were messed up, but sounds like it should be good. LGTM